src/app.py

```diff
@@ -963,7 +963,85 @@ def publish_datastage(identifier):
     except Exception as e:
         logger.error(e, exc_info=True)
         return Response("Unexpected error while creating a dataset: " + str(e) + "  Check the logs", 500)
-             
+
+
+@app.route('/datasets/metadata-json', methods=['PUT'])
+@secured(groups="HuBMAP-read")
+def datasets_metadata_json():
+    """
+    See publish_datastage() for additional information.
+
+    Use the Neo4J query (above) replacing ds.uuid with the value of ds.uuid, e.g.:
+    curl --verbose --request PUT \
+     --url ${INGESTAPI_URL}/datasets/metadata-json \
+     --header "Content-Type: application/json" \
+     --header "Accept: application/json" \
+     --header "Authorization: Bearer ${TOKEN}" \
+     --data '{"uuid":["fdacb21245e14fd126bfce05ae191857", "b4ed22922fe9efcc68bb656da04052cb"]}'
+    """
+    if not request.is_json:
+        return Response("json request required", 400)
+    dataset_uuids = request.json.get('uuid')
+    if type(dataset_uuids) is not list or len(dataset_uuids) == 0:
+        return Response("json request must contain a uuid key which is an array of dataset uuids", 400)
+
+    try:
+        auth_helper = AuthHelper.configured_instance(app.config['APP_CLIENT_ID'], app.config['APP_CLIENT_SECRET'])
+        ingest_helper = IngestFileHelper(app.config)
+        user_info = auth_helper.getUserInfoUsingRequest(request, getGroups=True)
+        if user_info is None:
+            return Response("Unable to obtain user information for auth token", 401)
+        if isinstance(user_info, Response):
+            return user_info
+
+        if 'hmgroupids' not in user_info:
+            return Response("User has no valid group information.", 403)
+        if data_admin_group_uuid not in user_info['hmgroupids']:
+            return Response("User must be a member of the HuBMAP Data Admin group.", 403)
+
+        for dataset_uuid in dataset_uuids:
+            q = f"MATCH (e:Dataset {{uuid: '{dataset_uuid}'}}) RETURN " \
```

Comment:
@shirey @ChuckKollar according to [the original card ](https://github.com/hubmapconsortium/manual-data-ingest/issues/256), it seems we should be calling Entity-API to fetch the information rather than relying on a neo4j query. We also won't have to call `string_helper.convert_str_literal()` to process the data. 

The PUT call to publish dataset is using neo4j query already, so I didn't raise up this question in PR #383 
What do you think?

Reply:
@yuanzhou I believe that I am just taking a segment of the code that is already in the "Publish" section of ingest-api and using it. I am just following that template, and it seems to work. If you want this to be done some other way, I suggest that you discuss this with @shirey as I have not worked with this micro-service much in the past.